public/app/plugins/datasource/azuremonitor/components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -167,12 +200,65 @@ const ResourcePicker = ({
       }
       setIsLoading(false);
     },
-    [loadInitialData, resourcePickerData, queryType]
+    [loadInitialData, resourcePickerData, queryType, filters]
+  );
+
+  const loadFilteredRows = useCallback(
+    async (filters: { subscriptions: string[]; types: string[]; locations: string[] }) => {
+      try {
+        setIsLoading(true);
+        const filteredRows = await resourcePickerData.fetchFiltered(queryType, filters);
+        setRows(filteredRows);
+      } catch (error) {
+        setErrorMessage(messageFromError(error));
+      }
+      setIsLoading(false);
+    },
+    [resourcePickerData, queryType]
   );
 
+  const updateFilters = (value: Array<ComboboxOption<string>>, filterType: 'subscriptions' | 'types' | 'locations') => {
+    const updatedFilters = { ...filters };
+    const values = value.map((v) => v.value);
+    switch (filterType) {
```

Comment:
this looks like it should be 
```
if (filterType === 'subscriptions') {
  updatedFilters.subscriptions = values;
```
}

Reply:
Resolving this as the switch statement is more applicable here 😊